cli/azd/pkg/project/project.go

```diff
@@ -205,7 +229,17 @@ func SaveConfig(ctx context.Context, config config.Config, projectFilePath strin
 
 // Saves the current instance back to the azure.yaml file
 func Save(ctx context.Context, projectConfig *ProjectConfig, projectFilePath string) error {
-	projectBytes, err := yaml.Marshal(projectConfig)
+	copy := *projectConfig
```

Comment:
This doesn't do a deep copy so the `Services` end up having their paths wacked, which was unintended - need to fix this.